src/Evolution/Systems/ScalarWave/EnergyDensity.hpp

```diff
@@ -0,0 +1,26 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace ScalarWave {
+/// @{
+/*!
+ * \brief Computes the energy density of the scalar wave system.
+ *
+ * Below is the fucntion used to calculate the energy density.
+ * \f{align*}
+ * \epsilon = \frac{1}{2}\left( \Pi^{2} + \abs{\Phi}^{2} \right)
+ * \f}
+ */
+template <size_t SpatialDim>
+void energy_density(
+    const gsl::not_null<Scalar<DataVector>*> result,
```

Comment:
in forward declarations you don't need `const` on variables that are taken by value. As in:

```cpp
void foo1(double a);
void foo2(const double& a);
void foo3(double& a);
```
the `&` makes the `const` important because without the const, you could modify the variable `a` in the callers scope. That is,
```cpp
const double b = 1.2;
foo1(b); // will never modify b because `b` is copied
foo2(b); // will never modify `b` because `b` is a `const &`
foo3(b); // might modify `b`, you don't know unless you look inside `foo3`
```